### ISSUE TITLE
Use full file info from implicit context

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode-copilot-release#2320]
Panel chat implicit context has always just included the current selection or the code that is currently in the viewport. We should try to include the full file, summarized if necessary, but draw the attention of the LLM to the selected/visible region.

**Comment by ThioJoe** [comment]
Yes I was just wondering this today. It would be nice to be able to talk to it and say something like "change this part to do X", but also let it have the full context of the file.  Right now selections seem to only give that text only, even with the `#file` variable:
<img src="https://github.com/user-attachments/assets/d54a0db6-f51a-4a9b-bffa-c13712ecbc14" alt="GitHub chat window" width="500">

That's with the current prerelease `v0.23.2024102903 (pre-release)`
